### DISCUSSION TITLE
How is v2 / vue3 coming along?

### DISCUSSION THREAD

**Discussion by adrianrudnik** [newcat/baklavajs#216] [General]
Hi there, just tried the v1 with Vue 3 and was also struck by the panning/zooming-issue and tried to v2 branch, but the docs are not yet there and I could not find any working example. Do you have any vague timeline for the nice approach I can see with v2?

**Comment by newcat** [comment]
I'd say v2 is about 95% complete, at least regarding the initial release (I have more features planned but they'll be released after the initial release). I am planning to release an official beta with some more documentation in the coming weeks. For now, you can try out the new version by installing `@baklavajs/core@next` (same for the other packages). The documentation can be found here: https://v2.baklava.tech/

**Comment by adrianrudnik** [comment]
Hey, that sounds awesome. I will try the new version later the upcoming week, thank you for the hint.

A side question though, as I only work with the visual part for a task based flow (one input to whatever), not the processing/engine: Coming from the current exported format, there is no direct indication if a `nodes[].interfaces` is an input or output correct? Trying to parse it with something else, not the baklava engine, seems to be a bit tricky here. Anything I'm missing or is it possible to expose additional details in the export?

Anyway, great work, I would love to see v2 and vue3. Mayber consider adding your lib to https://github.com/vuejs/awesome-vue for some traction!

**Reply by newcat** [reply]
First of all: thank you for the generous donation! There is a way to add data to a save state (although it is not yet documented) using hooks or by subclassing and overriding the `save` and `load` methods, as the [AdvancedNode](https://github.com/newcat/baklavajs/blob/master/packages/baklavajs-playground/src/AdvancedNode.ts) does it for example. But in v2 this is not neccessary anymore for your usecase, as the new [`INodeState`](https://v2.baklava.tech/api/interfaces/_baklavajs_core.INodeState.html) now has both an `inputs`, as well as an `outputs` property.
